### PR TITLE
HHH-13285 : ClassCastException: org.dom4j.DocumentFactory cannot be cast to org.dom4j.DocumentFactory after dom4j update (5.3)

### PR DESCRIPTION
**Pull request by gbadner**
https://hibernate.atlassian.net/browse/HHH-13285 (5.3)

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/internal/util/xml/XMLHelper.java [hunk 1]
```diff
@@ -22,23 +23,34 @@
 public final class XMLHelper {
 	private final DocumentFactory documentFactory;
 
-	public XMLHelper(ClassLoaderService classLoaderService) {
```
Comment (scottmarlow) [inline hibernate-core/src/main/java/org/hibernate/internal/util/xml/XMLHelper.java]: Do Hibernate Search or other Hibernate projects new up XMLHelper via XMLHelper(ClassLoaderService classLoaderService)?

Comment (scottmarlow) [reply hibernate-core/src/main/java/org/hibernate/internal/util/xml/XMLHelper.java]: Overall, the change looks good to me.

**Comment by gbadner** [comment]
Merged.

File: hibernate-core/src/main/java/org/hibernate/internal/util/xml/XMLHelper.java (hunk 1 above)
Comment (scottmarlow) [reply hibernate-core/src/main/java/org/hibernate/internal/util/xml/XMLHelper.java]: If there is concern about existing references to this internal class from other Hibernate projects, we could add XMLHelper(ClassLoaderService) ctor back in.

**Pull request closed without merging**